portal/models/assessment_status.py

```diff
@@ -186,6 +190,37 @@ def overall_status(self):
                 result = 'In Progress'
         return result
 
+    def withdrawn(self):
+        """Determine if user has `withdrawn` from study """
+        # No easy tie between QB and organization.  Make assumption
+        # that if the QBD isn't intervention tied, the user is associated
+        # with the intervention by org.
+        intervention_qbs = QuestionnaireBank.query.filter(
+            QuestionnaireBank.intervention_id.isnot(None))
+        if self.qbd.questionnaire_bank in intervention_qbs:
+            return False
+
+        # With multiple root organizations, the consent lookup would
+        # be indeterminate - don't allow
+        root_orgs = OrgTree().find_top_level_org(self.user.organizations)
+        if len(root_orgs) > 1:
+            current_app.logger.error(
```

Comment:
Should you also exit or throw an exception at this point (to prevent the lookup)?

Reply:
*shrug* - perhaps, but then we risk bringing everything to a halt when processing the hourly reminder tasks.